File: autotts_reference/decompiled_java/com/vnspeak/autotts/GetSampleText.java
/*
 * Decompiled with CFR 0.152.
 * 
 * Could not load the following classes:
 *  android.app.Activity
 *  android.content.Intent
 *  android.os.Bundle
 */
package com.vnspeak.autotts;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import c3.y;
import java.util.Locale;

public class GetSampleText
extends Activity {
    public static Locale a(Intent object) {
        if (object != null && (object = object.getStringExtra("language")) != null) {
            return new Locale((String)object);
        }
        return Locale.getDefault();
    }

    public void onCreate(Bundle object) {
        super.onCreate((Bundle)object);
        Locale locale = GetSampleText.a(this.getIntent());
        String string = y.a(locale.getISO3Language());
        object = string;
        if (string.isEmpty()) {
            object = new StringBuilder();
            ((StringBuilder)object).append("Sorry. Sample text for language ");
            ((StringBuilder)object).append(locale.getDisplayName(new Locale("eng")));
            ((StringBuilder)object).append(" is missing.");
            object = ((StringBuilder)object).toString();
        }
        string = new Intent();
        string.putExtra("sampleText", (String)object);
        this.setResult(0, (Intent)string);
        this.finish();
    }
}

File: app/src/main/java/com/tts/easyvoice/GetSampleText.kt
package com.tts.easyvoice
import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.speech.tts.TextToSpeech
import java.util.Locale
class GetSampleText : Activity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val langExtra = intent?.getStringExtra("language")
        val locale = if (langExtra != null) Locale(langExtra) else Locale.getDefault()
        // isO3Language THROWS MissingResourceException for a language with no
        // three-letter code -- measured, not assumed: Locale("xx").isO3Language
        // is "Couldn't find 3-letter language code for xx". This activity is
        // exported and its "language" extra comes from ANOTHER app, so an
        // unguarded read is a crash any caller can trigger. AutoTTS's
        // GetSampleText reads it bare as well; this is the same "our refactor is
        // not the point, the phone crashing is" split as the API-26 guards, and
        // the fallback is the sentence the screen already shows for a language
        // it has no sample for. A valid code behaves exactly as before.
        val sample = try { SampleTexts.get(locale.isO3Language) } catch (_: Exception) { "" }
        val text = if (sample.isEmpty()) "Sorry. Sample text for language ${locale.getDisplayName(Locale("eng"))} is missing." else sample
        val data = Intent()
        data.putExtra("sampleText", text)
        setResult(TextToSpeech.LANG_AVAILABLE, data)
        finish()
    }
}
